### PR TITLE
OCSADV-399-G

### PR DESCRIPTION
**Pull request by tpolecat**
Prepare StrehlFeature for possibly-missing coordinates. (4/6)

This code is used by the TPE drawing code, which catches exceptions and shows them to the user. So I'm hoping the exception added to `targetListToStarList` is good enough.

### CODE REVIEW

File: bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java [hunk 1]
```diff
@@ -407,30 +408,41 @@ private BandsList getBandpass(GuideProbe.Type type) {
...
+        return
+            target.getTarget().getRaDegrees(when).flatMap(ra ->
+            target.getTarget().getDecDegrees(when).map(dec -> {
+                String name = target.getTarget().getName();
+                double baseX = _tii.getBasePos().getRaDeg();
+                double baseY = _tii.getBasePos().getDecDeg();
+                Magnitude undef = new Magnitude(Magnitude.Band.J, MascotConf.invalidMag());
+                double bmag = target.getTarget().getMagnitude(Magnitude.Band.B).getOrElse(undef).getBrightness();
+                double vmag = target.getTarget().getMagnitude(Magnitude.Band.V).getOrElse(undef).getBrightness();
+                double rmag = target.getTarget().getMagnitude(Magnitude.Band.R).getOrElse(undef).getBrightness();
```
Comment (cquiroz) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: This is unrelated to this particular PR but I think this should follow the "use one of the R-bands" logic

File: bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java [hunk 2]
```diff
@@ -407,30 +408,41 @@ private BandsList getBandpass(GuideProbe.Type type) {
...
+    // throws an exception if a star's coordinates are unknown
     private Star[] targetListToStarList(List<SPTarget> targetList) {
         Star[] starList = new Star[3];
         int n = targetList.size();
         for (int i = 0; i < starList.length; i++) {
-            starList[i] = i < n ? targetToStar(targetList.get(i)) : null;
+            if (i < n) {
+                final SPTarget spt = targetList.get(i);
+                final Option<Star> op = targetToStar(spt);
+                if (op.isEmpty()) {
```
Comment (cquiroz) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: When would this happen in practice?

Comment (tpolecat) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: I don't know, probably never. That's why I'm not too worried about throwing the exception above.

**Comment by cquiroz** [comment]
The code looks good, A general worry is how introducing the scheduling block is going to affect tests, Have you been running tests with your PR code? I worry about the AGS tests for example

**Comment by tpolecat** [comment]
@cquiroz yes, I run all the tests for each PR and they all pass.

File: bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java (hunk 1 above)
Comment (tpolecat) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: Ok if you want to make a PR to this branch I can merge it in … or just open a new issue?

Comment (cquiroz) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: I think a new issue is the right approach, I'm not sure how can we do this easily on Java-oldModel-land

Comment (cquiroz) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java]: Added OCSADV-407

**Pull request merged**